core/rpc-server/src/lib.rs

```diff
@@ -35,37 +27,15 @@ mod mock;
 #[cfg(test)]
 mod tests;
 
-pub async fn run_server<Enclave, FetchSidechainBlocks>(
+pub async fn run_server<FetchSidechainBlocks>(
 	addr: impl ToSocketAddrs,
-	enclave: Arc<Enclave>,
 	sidechain_block_fetcher: Arc<FetchSidechainBlocks>,
 ) -> anyhow::Result<SocketAddr>
 where
-	Enclave: DirectRequest,
 	FetchSidechainBlocks: FetchBlocks<SignedBlock> + Send + Sync + 'static,
 {
 	let mut server = WsServerBuilder::default().build(addr).await?;
 
-	// FIXME: import block should be moved to trusted side.
-	let mut import_sidechain_block_module = RpcModule::new(enclave);
-	import_sidechain_block_module.register_method(
-		RPC_METHOD_NAME_IMPORT_BLOCKS,
-		|params, enclave| {
-			debug!("{} params: {:?}", RPC_METHOD_NAME_IMPORT_BLOCKS, params);
-
-			let enclave_req = RpcRequest::compose_jsonrpc_call(
-				RPC_METHOD_NAME_IMPORT_BLOCKS.into(),
-				vec![params.one::<Vec<SignedBlock>>()?.to_hex()],
-			)
-			.unwrap();
-
-			enclave
-				.rpc(enclave_req.as_bytes().to_vec())
-				.map_err(|e| CallError::Failed(e.into()))
-		},
-	)?;
-	server.register_module(import_sidechain_block_module).unwrap();
-
 	let fetch_sidechain_blocks_module = BlockFetchServerModuleBuilder::new(sidechain_block_fetcher)
 		.build()
 		.map_err(|e| CallError::Failed(e.to_string().into()))?; // `to_string` necessary due to no all errors implementing Send + Sync.
```

Comment:
hmmm, I guess the sidechain block fetcher should also be on the trusted side eventually.